content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+
+The typical uses for `warn` are:
+
+* `warn` at the same level but a different version (e.g. pin `enforce` to *restricted+v1.23* and `warn` at *restricted+latest*)
+* `warn` at a stricter level (e.g. `enforce` baseline, `warn` restricted)
+
+
+Let's create 3 namespaces and apply all 3 PSA security profiles. First, create the namespaces:
+
+```shell
+kubectl create namespace kube-test-privileged
+kubectl create namespace kube-test-baseline
+kubectl create namespace kube-test-restricted
+```
+
+Add the respective labels.
+
+```shell
+# kube-test-privileged enforces a "privileged" security policy and warns / audits on baseline
+kubectl label --overwrite ns kube-test-privileged \
+  pod-security.kubernetes.io/enforce=privileged \
+  pod-security.kubernetes.io/warn=baseline \
+  pod-security.kubernetes.io/audit=baseline
+
+# kube-test-baseline enforces a "baseline" security policy and warns / audits on restricted
+kubectl label --overwrite ns kube-test-baseline \
+  pod-security.kubernetes.io/enforce=baseline \
+  pod-security.kubernetes.io/warn=restricted \
+  pod-security.kubernetes.io/audit=restricted
+
+# kube-test-restricted enforces a "restricted" security policy and audits on restricted
+kubectl label --overwrite ns kube-test-restricted \
+  pod-security.kubernetes.io/enforce=restricted \
+  pod-security.kubernetes.io/audit=restricted
+```
+
+### Deploy demo workloads
+
+Each workload represents a higher level of security that would not pass the profile that comes after it.
+
+**Privileged level and workload**
+
+For the privileged pod, allow privilege escalation. This allows the process inside a container to gain new processes and can be dangerous if untrusted ([more info](https://kubernetes.io/docs/concepts/security/pod-security-standards/#privileged)).
+
+First, let's try to deploy it in the restricted namespace:
+
+```shell
+cat <<EOF | kubectl -n kube-test-restricted apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+The output is similar to this:
+
+```shell
+Error from server (Forbidden): error when creating "STDIN": pods "nginx-privileged" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "nginx" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+```
+
+Now let's try the `kube-test-privileged` namespace
+
+```shell
+cat <<EOF | kubectl -n kube-test-privileged apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
```

Comment:
nit: I'd prefer to use the pause image (`k8s.gcr.io/pause`) in examples that don't need to actually run anything.

It's less likely to terminate for other reasons, and it's an image we own (not promoting other projects)

Reply:
Maybe add a comment about why we picked it?